ols/app/models/config.py

```diff
@@ -231,6 +231,31 @@ def __init__(self, data: Optional[dict] = None) -> None:
         if data is None:
             return
         self.name = data.get("name", None)
+
+        self.set_provider_type(data)
+        self.url = data.get("url", None)
+        self.credentials = _get_attribute_from_file(data, "credentials_path")
+
+        # OLS-622: Provider-specific configuration parameters in olsconfig.yaml
+        self.project_id = data.get("project_id", None)
+        if self.type == constants.PROVIDER_WATSONX and self.project_id is None:
+            raise InvalidConfigurationError(
+                f"project_id is required for Watsonx provider {self.name}"
+            )
+
+        self.setup_models_config(data)
```

Comment:
why is this call before the azure validation check?

Reply:
(I would have expected to do all the provider config type checking first, then load the model config data)